bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/UpnpDiscoveryService.java

```diff
@@ -102,6 +105,11 @@ protected void startScan() {
         }
         upnpService.getRegistry().addListener(this);
         upnpService.getControlPoint().search();
+
+        // Belkin WeMo devices can not be discovered by default search() function, but have to be searched by their
+        // named service.
+        ServiceType wemoBasicEvent = new ServiceType("Belkin", "basicevent");
+        upnpService.getControlPoint().search(new ServiceTypeHeader(wemoBasicEvent));
```

Comment:
I would prefer to use a generic mechanism. For example as @kaikreuzer already stated by extending the UPnPDiscoveryParticipant.